### PR TITLE
forgot to make a branch, changed a comment

### PR DESCRIPTION
**Pull request by andrayantelo**

### CODE REVIEW

File: sortandsearch.c [hunk 1]
```diff
@@ -0,0 +1,128 @@
+#include <stdio.h>
+//#include <cs50.h>
+#include <stdbool.h>
+
+void sort(int values[], int n);
+bool search(int value, int values[], int n);
+
+int main(void) {
+    int values[4];
```
Comment (johntyree) [inline sortandsearch.c:9]: I think you can do `int values[] = {39, 40, 41, 42};`

File: sortandsearch.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        printf("Found the value\n");
+    }
+    else {
+        printf("Didn't find it: %d\n", found);
+    }
+}
+
+
+
+void sort(int values[], int n)
```
Comment (johntyree) [inline sortandsearch.c:37]: If you make some helper functions, this function will be much shorter and easier to read.

```
find_smaller_between(int values[], int value, int start, int end);
swap(int values[], int i, int j);
etc..
```

File: sortandsearch.c [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        }
+    }
+}
+
+bool search(int value, int values[], int n)
+{
+    // TODO: implement a searching algorithm
+    // Return false if n is non-positive
+    if (n < 0) {
```
Comment (johntyree) [inline sortandsearch.c:77]: `0` is non-positive...

File: sortandsearch.c [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        return false;
+    }
+    
+    // Base case. 
+    if (n == 1) {
+        if (value ==  values[0]) {
```
Comment (johntyree) [inline sortandsearch.c:83]: Just return the result of the conditional.

File: sortandsearch.c [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    
+    // find midpoint
+    int midpoint = values[n/2];
+    if (value == midpoint) {
```
Comment (johntyree) [inline sortandsearch.c:93]: If you ignore this case, will it be handled and finish correctly on the next recursive call?

File: sortandsearch.c [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    int j;
+    // Check if value is to the left of midpoint
+    if (value < midpoint) {
+      
+        // if it is, then throw away right half, get the left half, and
+        // search through it
+
+        // get left half
+       
+        int values_copy[n/2 - 1];
```
Comment (johntyree) [inline sortandsearch.c:107]: Why do you need to copy the array?

**Review by johntyree — CHANGES_REQUESTED** [review]

File: sortandsearch.c (hunk 1 above)
Comment (andrayantelo) [reply sortandsearch.c:9]: oh. lol :heart:

File: sortandsearch.c (hunk 2 above)
Comment (andrayantelo) [reply sortandsearch.c:37]: I haven't gone and cleaned up sort yet, just search

Comment (andrayantelo) [reply sortandsearch.c:37]: But yes, noted, will do when I start cleaning it up.

File: sortandsearch.c (hunk 6 above)
Comment (andrayantelo) [reply sortandsearch.c:107]: I don't. I fixed it later. Before I saw this comment. :)

**Pull request merged**